### ISSUE TITLE
Dockerfile needs compiler on aarch64

### ISSUE THREAD

**Issue by JannKleen** [reflex-dev/reflex#1556]
**Describe the bug**
On an ARM64 systems, psutil (and potentially python-multipart?) aren't provided as binary packages and need to be compiled. This won't work with the default docker file provided.

**To Reproduce**
On an ARM64 system (in this case an AWS EC2 t4g.micro instance) run:
```shell
docker build -t reflex-project:latest .
```

**Expected behavior**
The build should succeed. If you add the following before running pip install, it works:
```
RUN apt-get update -y && \
    apt-get install -y --no-install-recommends build-essential python3-dev && \
    rm -rf /var/lib/apt/lists/* /var/cache/apt/* /var/log/dpkg.log
```

**Console logs**
```
11.44 Building wheels for collected packages: psutil, python-multipart
11.45   Building wheel for psutil (pyproject.toml): started
11.76   Building wheel for psutil (pyproject.toml): finished with status 'error'
11.77   error: subprocess-exited-with-error
11.77   
11.77   × Building wheel for psutil (pyproject.toml) did not run successfully.
11.77   │ exit code: 1
11.77   ╰─> [43 lines of output]
11.77       running bdist_wheel
11.77       running build
11.77       running build_py
11.77       creating build
11.77       creating build/lib.linux-aarch64-cpython-311
11.77       creating build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_common.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-311/psutil
11.77       creating build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
11.77       running build_ext
11.77       building 'psutil._psutil_linux' extension
11.77       creating build/temp.linux-aarch64-cpython-311
11.77       creating build/temp.linux-aarch64-cpython-311/psutil
11.77       gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=595 -DPy_LIMITED_API=0x03060000 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -DPSUTIL_LINUX=1 -I/app/venv/include -I/usr/local/include/python3.11 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-311/psutil/_psutil_common.o
11.77       psutil could not be installed from sources because gcc is not installed. Try running:
11.77         sudo apt-get install gcc python3-dev
11.77       error: command 'gcc' failed: No such file or directory
11.77       [end of output]
11.77   
11.77   note: This error originates from a subprocess, and is likely not a problem with pip.
11.77   ERROR: Failed building wheel for psutil
11.77   Building wheel for python-multipart (setup.py): started
12.23   Building wheel for python-multipart (setup.py): finished with status 'done'
12.23   Created wheel for python-multipart: filename=python_multipart-0.0.5-py3-none-any.whl size=31666 sha256=25be4f9b06a5b09ffaabb578274a0c2b013ac0d769c2cc83bb72a99d92cde042
12.23   Stored in directory: /root/.cache/pip/wheels/3a/d4/d0/dbbf3c534e9c5dfbdb2729d10bd12548b8a07103030c07d9e7
12.23 Successfully built python-multipart
12.23 Failed to build psutil
12.23 ERROR: Could not build wheels for psutil, which is required to install pyproject.toml-based projects
```

**Specifics (please complete the following information):**
 - Python Version: 3.11-slim
 - Reflex Version: 0.2.4
 - OS: Host is Debian bookworm

**Comment by masenf** [comment]
This should be fixed in 0.2.5 `docker-example`, check it out!

**Comment by JannKleen** [comment]
Yes, that works fine now! Thanks!
